api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -144,6 +149,27 @@ public DbUser updateUserWithRetries(
     while (true) {
       dbUser = userModifier.apply(dbUser);
       dbUser = accessSyncService.updateUserAccessTiers(dbUser, agent);
+
+      boolean enableInitialCreditsExpiration =
```

Comment:
Main logic for credit expiration is in this block

Reply:
does this mean when we roll out the change, existing users will have an expiration date depending on when they trigger `updateUserWithRetries`?

Reply:
Good call. Yes, that is what it was doing. I just updated it, so it only runs when a user gains access to their first user tier. Thanks for catching this.

Reply:
shall we also add some unit test to validate the logic is correct? like the expiration is set to proper value

Reply:
Good call. Tests have been moved to api/src/test/java/org/pmiops/workbench/access/AccessSyncServiceTest.java